### PR TITLE
Migration to gogoproto

### PR DESCRIPTION
**Pull request by ansrivas**
### IMPORTANT NOTE:
Tests are not passing currently. Turns out that `porpackage_test.go:48` is failing where we are trying to create `NewPorPackage` with a given transaction. The test is trying to fetch `HeightByBlockHash` which is failing currently.

Although this work has been mostly a replacement of `protobuf` with `gogoprotobuf` I'd need some guidance in testing this.
@yilunzhang @doraemon0 

### Proposed changes in this pull request
protobuf has been replaced with gogoprotobuf.
* All the proto files have been migrated to use v3 of gogoproto.
* Definitions are generated by `protoc v3.6.0`
* gogoproto provides an option to generate test files, so this is being used.
* `SigAlgo_ECDSA` has been renamed simply to `ECDSA` ( as generated by gogoproto)
*  Renamed `golang/protobuf` -> `gogo/protobuf` in all the imports

Protobuf definitions are generated as follows:
Dependencies:
```bash
go get github.com/gogo/protobuf/protoc-gen-gofast
go get github.com/gogo/protobuf/proto
go get github.com/gogo/protobuf/protoc-gen-gogoslick
go get github.com/gogo/protobuf/gogoproto
```
And definitions are generated using `protoc v3.6.0`
```bash
protoc -I=. -I=$GOPATH/src -I=$GOPATH/src/github.com/gogo/protobuf/protobuf --gogoslick_out=. messages.proto 

protoc -I=. -I=$GOPATH/src -I=$GOPATH/src/github.com/gogo/protobuf/protobuf --gogoslick_out=. sigchain.proto 

protoc -I=. -I=$GOPATH/src -I=$GOPATH/src/github.com/gogo/protobuf/protobuf --gogoslick_out=. porpackage.proto
```

### Type (put an `x` where ever applicable)
- [x] Bug fix: Link to the issue #247 
- [x] Feature (Non-breaking change) 
- [ ] Feature (Breaking change)
- [ ] Documentation Improvement

### Checklist
Please put an `x` against the checkboxes. Write a small comment explaining if its `N/A` (not applicable)

- [x] Read the [CONTRIBUTION guidelines](https://github.com/nknorg/nkn#contributing).
- [ ] All the tests are passing after the introduction of new changes. ( No, see above )
- [x] Added tests respective to the part of code I have written.
- [x] Added proper documentation where ever applicable (in code and README.md).
- [x] Code has been written according to [NKN-Golang-Style-Guide](https://github.com/nknorg/nkn/wiki/NKN-Golang-Style-Guide)

### Extra information
Any extra information related to this pull request.

### CODE REVIEW

File: api/websocket/client/messages.proto [hunk 1]
```diff
@@ -2,12 +2,26 @@ syntax = "proto3";
...
+
+// For tests
+option (gogoproto.testgen_all) = true;
+option (gogoproto.equal_all) = true;
+option (gogoproto.populate_all) = true;
+
 message OutboundMessage {
-  string dest = 1;
-  bytes payload = 2;
+  string dest = 1 ;
```
[the remark this answers is not in the document]
Comment (ansrivas) [reply api/websocket/client/messages.proto]: I was adding few more optimizations to each field, but removed it later.
Will remove the spaces.

**Comment by ansrivas** [comment]
@yilunzhang which version of `protoc` compiler you are using? I generated the definitions using this:
```
 $  protoc --version
libprotoc 3.6.0
```
Can you also post the difference? I can try and check with different versions of the compilers.

**Comment by yilunzhang** [comment]
You are right, I was using 3.5.1. Let's stick to your version. Here is the diff for /por/porpackage.pb.go in case you still want it:

-       return fileDescriptor_porpackage_2d5a03740e4e34c1, []int{0}
+       return fileDescriptor_porpackage_09558aaa330d79a1, []int{0}

-func init() { proto.RegisterFile("porpackage.proto", fileDescriptor_porpackage_2d5a03740e4e34c1) }
+func init() { proto.RegisterFile("porpackage.proto", fileDescriptor_porpackage_09558aaa330d79a1) }
 
-var fileDescriptor_porpackage_2d5a03740e4e34c1 = []byte{
+var fileDescriptor_porpackage_09558aaa330d79a1 = []byte{

**Comment by yilunzhang** [comment]
I was trying to merge but get blocked even if I resolve the conflict. It says:

This branch cannot be rebased due to conflicts
Rebasing the commits of this branch on top of the base branch cannot be performed automatically due to conflicts encountered while reapplying the individual commits from the head branch.

Could you please do a "git pull --rebase upstream master" locally and push again? Here I assume upstream is the original nkn repo not your fork.

**Comment by dreamfly281** [comment]
> I was trying to merge but get blocked even if I resolve the conflict. It says:
> 
> This branch cannot be rebased due to conflicts
> Rebasing the commits of this branch on top of the base branch cannot be performed automatically due to conflicts encountered while reapplying the individual commits from the head branch.
> 
> Could you please do a "git pull --rebase upstream master" locally and push again? Here I assume upstream is the original nkn repo not your fork.

Maybe we can merge this  patch to Dev branch for temporary

**Comment by yilunzhang** [comment]
gogoprotobuf is mutually compatible with protobuf. I've tested this PR without any problem

**Comment by ansrivas** [comment]
@yilunzhang @dreamfly281  I have pushed a few commits after rebasing from nkn-master. I hope this will be in a state to merge.

**Comment by yilunzhang** [comment]
This time it becomes "This branch cannot be rebased due to conflicts" and I cannot resolve it 😂

**Comment by ansrivas** [comment]
Ahh, okay, let's do this. I will send a fresh request later today. :smile: 
(I have more or less everything documented in this pull request.)

**Pull request closed without merging**